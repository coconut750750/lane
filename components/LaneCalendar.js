import React, { Component } from 'react';
import { CalendarList } from 'react-native-calendars';
var _ = require('lodash');

import Colors from '../constants/Colors'

export default class LaneCalendar extends Component {
    constructor(props) {
        super(props);
        this.periodMark = props.markings;
        this.state = {
            allMark: props.markings
        };
    }

    onDayPress(day) {
        var selectedMark = {[day.dateString]: {selected: true, disableTouchEvent: true}};
        var periodMark = _.cloneDeep(this.periodMark);
        
        this.setState({
            allMark: _.merge(periodMark, selectedMark)
        });
    }

    render() {
        return (
            <CalendarList
                monthFormat={'MMMM yyyy'}
                onDayPress={ (day) => { this.onDayPress(day); } }
                markingType={'multi-period'}
                markedDates={{...this.state.allMark}}
                horizontal={true}
                pagingEnabled={true}
                theme={{
                    backgroundColor: Colors.background,
                    calendarBackground: Colors.background,
                    textSectionTitleColor: '#b6c1cd',
                    selectedDayBackgroundColor: Colors.primaryAlt,
                    selectedDayTextColor: 'white',
                    todayTextColor: Colors.primary,
                    dayTextColor: Colors.darkGray,
                    textDisabledColor: Colors.lightGray,
                    dotColor: Colors.primaryAlt,
                    selectedDotColor: '#ffffff',
                    arrowColor: Colors.primaryAlt,
                    monthTextColor: Colors.primary,
                    indicatorColor: Colors.secondary,
                    textDayFontFamily: 'roboto',
                    textMonthFontFamily: 'roboto-medium',
                    textDayHeaderFontFamily: 'roboto',
                    textDayFontWeight: '300',
                    textMonthFontWeight: '300',
                    textDayHeaderFontWeight: '300',
                    textDayFontSize: 16,
                    textMonthFontSize: 20,
                    textDayHeaderFontSize: 16,
                }}
                style={{
                    height: 0
                }}
            />
        );
    }
}