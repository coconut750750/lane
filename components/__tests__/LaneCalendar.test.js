import React from 'react'
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LaneCalendar from '../LaneCalendar'

Enzyme.configure({ adapter: new Adapter() });

describe('LaneCalendar', () => {
    let markings = {
        '2019-05-16': {
            periods: [
                { startingDay: true, endingDay: false, color: '#5f9ea0' },
                { startingDay: true, endingDay: false, color: '#ffa500' },
            ]
        }, '2019-05-17': {
            periods: [
                { startingDay: false, endingDay: true, color: '#5f9ea0' },
                { startingDay: false, endingDay: true, color: '#ffa500' },
            ]
        }
    };

    it('render without markings', () => {
        const component = shallow(<LaneCalendar markings={{}}/>);
        expect(component).toMatchSnapshot();
    });

    it('render with markings', () => {
        const component = shallow(<LaneCalendar markings={markings}/>);
        expect(component).toMatchSnapshot();
    });

    it('selected date is highlighted', () => {
        const component = shallow(<LaneCalendar markings={{}}/>);
        const instance = component.instance();
        instance.onDayPress({dateString: '2019-05-20'});
        expect(instance.state.allMark).toEqual({
            ['2019-05-20']: {selected: true, disableTouchEvent: true}
        });
    });

    it('selected marked date is highlighted', () => {
        const component = shallow(<LaneCalendar markings={markings}/>);
        const instance = component.instance();
        instance.onDayPress({dateString: '2019-05-16'});
        expect(instance.state.allMark['2019-05-16']).toEqual({
            periods: [
                { startingDay: true, endingDay: false, color: '#5f9ea0' },
                { startingDay: true, endingDay: false, color: '#ffa500' },
            ],
            selected: true,
            disableTouchEvent: true
        });
    });
});