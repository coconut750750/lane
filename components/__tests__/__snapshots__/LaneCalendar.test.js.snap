// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaneCalendar render with markings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LaneCalendar
    markings={
      Object {
        "2019-05-16": Object {
          "periods": Array [
            Object {
              "color": "#5f9ea0",
              "endingDay": false,
              "startingDay": true,
            },
            Object {
              "color": "#ffa500",
              "endingDay": false,
              "startingDay": true,
            },
          ],
        },
        "2019-05-17": Object {
          "periods": Array [
            Object {
              "color": "#5f9ea0",
              "endingDay": true,
              "startingDay": false,
            },
            Object {
              "color": "#ffa500",
              "endingDay": true,
              "startingDay": false,
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "markedDates": Object {
        "2019-05-16": Object {
          "periods": Array [
            Object {
              "color": "#5f9ea0",
              "endingDay": false,
              "startingDay": true,
            },
            Object {
              "color": "#ffa500",
              "endingDay": false,
              "startingDay": true,
            },
          ],
        },
        "2019-05-17": Object {
          "periods": Array [
            Object {
              "color": "#5f9ea0",
              "endingDay": true,
              "startingDay": false,
            },
            Object {
              "color": "#ffa500",
              "endingDay": true,
              "startingDay": false,
            },
          ],
        },
      },
      "markingType": "multi-period",
      "monthFormat": "MMMM yyyy",
      "onDayPress": [Function],
      "theme": Object {
        "arrowColor": "#3700b3",
        "backgroundColor": "#fff",
        "calendarBackground": "#fff",
        "dayTextColor": "#888",
        "dotColor": "#3700b3",
        "indicatorColor": "#03dac6",
        "monthTextColor": "#6200ee",
        "selectedDayBackgroundColor": "#3700b3",
        "selectedDayTextColor": "white",
        "selectedDotColor": "#ffffff",
        "textDayFontFamily": "roboto",
        "textDayFontSize": 16,
        "textDayFontWeight": "300",
        "textDayHeaderFontFamily": "roboto",
        "textDayHeaderFontSize": 16,
        "textDayHeaderFontWeight": "300",
        "textDisabledColor": "#ccc",
        "textMonthFontFamily": "roboto-medium",
        "textMonthFontSize": 20,
        "textMonthFontWeight": "300",
        "textSectionTitleColor": "#b6c1cd",
        "todayTextColor": "#6200ee",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "markedDates": Object {
          "2019-05-16": Object {
            "periods": Array [
              Object {
                "color": "#5f9ea0",
                "endingDay": false,
                "startingDay": true,
              },
              Object {
                "color": "#ffa500",
                "endingDay": false,
                "startingDay": true,
              },
            ],
          },
          "2019-05-17": Object {
            "periods": Array [
              Object {
                "color": "#5f9ea0",
                "endingDay": true,
                "startingDay": false,
              },
              Object {
                "color": "#ffa500",
                "endingDay": true,
                "startingDay": false,
              },
            ],
          },
        },
        "markingType": "multi-period",
        "monthFormat": "MMMM yyyy",
        "onDayPress": [Function],
        "theme": Object {
          "arrowColor": "#3700b3",
          "backgroundColor": "#fff",
          "calendarBackground": "#fff",
          "dayTextColor": "#888",
          "dotColor": "#3700b3",
          "indicatorColor": "#03dac6",
          "monthTextColor": "#6200ee",
          "selectedDayBackgroundColor": "#3700b3",
          "selectedDayTextColor": "white",
          "selectedDotColor": "#ffffff",
          "textDayFontFamily": "roboto",
          "textDayFontSize": 16,
          "textDayFontWeight": "300",
          "textDayHeaderFontFamily": "roboto",
          "textDayHeaderFontSize": 16,
          "textDayHeaderFontWeight": "300",
          "textDisabledColor": "#ccc",
          "textMonthFontFamily": "roboto-medium",
          "textMonthFontSize": 20,
          "textMonthFontWeight": "300",
          "textSectionTitleColor": "#b6c1cd",
          "todayTextColor": "#6200ee",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`LaneCalendar render without markings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LaneCalendar
    markings={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "markedDates": Object {},
      "markingType": "multi-period",
      "monthFormat": "MMMM yyyy",
      "onDayPress": [Function],
      "theme": Object {
        "arrowColor": "#3700b3",
        "backgroundColor": "#fff",
        "calendarBackground": "#fff",
        "dayTextColor": "#888",
        "dotColor": "#3700b3",
        "indicatorColor": "#03dac6",
        "monthTextColor": "#6200ee",
        "selectedDayBackgroundColor": "#3700b3",
        "selectedDayTextColor": "white",
        "selectedDotColor": "#ffffff",
        "textDayFontFamily": "roboto",
        "textDayFontSize": 16,
        "textDayFontWeight": "300",
        "textDayHeaderFontFamily": "roboto",
        "textDayHeaderFontSize": 16,
        "textDayHeaderFontWeight": "300",
        "textDisabledColor": "#ccc",
        "textMonthFontFamily": "roboto-medium",
        "textMonthFontSize": 20,
        "textMonthFontWeight": "300",
        "textSectionTitleColor": "#b6c1cd",
        "todayTextColor": "#6200ee",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "markedDates": Object {},
        "markingType": "multi-period",
        "monthFormat": "MMMM yyyy",
        "onDayPress": [Function],
        "theme": Object {
          "arrowColor": "#3700b3",
          "backgroundColor": "#fff",
          "calendarBackground": "#fff",
          "dayTextColor": "#888",
          "dotColor": "#3700b3",
          "indicatorColor": "#03dac6",
          "monthTextColor": "#6200ee",
          "selectedDayBackgroundColor": "#3700b3",
          "selectedDayTextColor": "white",
          "selectedDotColor": "#ffffff",
          "textDayFontFamily": "roboto",
          "textDayFontSize": 16,
          "textDayFontWeight": "300",
          "textDayHeaderFontFamily": "roboto",
          "textDayHeaderFontSize": 16,
          "textDayHeaderFontWeight": "300",
          "textDisabledColor": "#ccc",
          "textMonthFontFamily": "roboto-medium",
          "textMonthFontSize": 20,
          "textMonthFontWeight": "300",
          "textSectionTitleColor": "#b6c1cd",
          "todayTextColor": "#6200ee",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
